fsm/observer.go

```
@@ -130,7 +158,12 @@ func (s *CachedObserver) WaitForState(ctx context.Context,
 		return NewErrWaitingForStateTimeout(
 			state, s.lastNotification.NextState,
 		)
-	case <-ch:
+	case errorOccured := <-ch:
+		if errorOccured {
+			return NewErrWaitingForStateEarlyAbortError(
```

Could we just return the last error that made the fsm abort?